mocks_test.go

```diff
@@ -95,6 +95,25 @@ func (m *mockGauge) With(labelValues ...string) metrics.Gauge {
 	return args.Get(0).(metrics.Gauge)
 }
 
+// // mockHistogram provides the mock implementation of the metrics.Histogram object
+type mockHistogram struct {
+	mock.Mock
+}
+
+func (m *mockHistogram) Observe(value float64) {
+	m.Called(value)
+}
+
+func (m *mockHistogram) With(labelValues ...string) metrics.Histogram {
+	for _, v := range labelValues {
+		if !utf8.ValidString(v) {
+			panic("not UTF-8")
+		}
+	}
+	args := m.Called(labelValues)
+	return args.Get(0).(metrics.Histogram)
```

Comment:
```suggestion
	return m
```
I don't think we'll want something special returned here, so the easiest thing to do is return our same mock - then the same mock can expect `With()` and `Observe()` calls.